### PR TITLE
feat: Add complete sign up flow

### PR DESCRIPTION
**Pull request by blystsiv**
- Implement signup form with validation and submission
- Show error alert for existing users
- Redirect to login page on error
- Update local storage with registered users

### CODE REVIEW

File: src/pages/Signup/Signup.tsx [hunk 1]
```diff
@@ -1,21 +1,76 @@
...
+    watch,
     formState: { errors },
-  } = useForm();
+  } = useForm<User>();
+
+  useEffect(() => {
+    const storedUsers = localStorage.getItem('registeredUsers');
+
+    if (storedUsers) {
+      setRegisteredUsers(JSON.parse(storedUsers));
```
Comment (kl2karpenko) [inline src/pages/Signup/Signup.tsx]: add here try/catch to get if any errors that can happen on parce

File: src/pages/Signup/Signup.tsx [hunk 2]
```diff
@@ -1,21 +1,76 @@
...
+    }
+  }, []);
+
+  const handleFormSubmit = (data: User) => {
+    try {
+      const isEmailAlreadyUsed = registeredUsers.some(
```
Comment (kl2karpenko) [inline src/pages/Signup/Signup.tsx]: replace with only 1 mapping

```
      let isEmailAlreadyUsed = false;
      let isUsernameAlreadyUsed = false;

      registeredUsers.forEach(
        (user) => {
             if (user.email === data.email) { isEmailAlreadyUsed = true; }
             if (user.username === data.username) { isUsernameAlreadyUsed = true; }
        
        },
      );
```

File: src/pages/Signup/Signup.tsx [hunk 3]
```diff
@@ -56,13 +109,32 @@ export const Signup = () => {
...
+                }
+                if (!/[A-Z]/.test(val)) {
+                  return 'Password must contain at least one uppercase letter';
+                }
+                if (!/[a-z]/.test(val)) {
+                  return 'Password must contain at least one lowercase letter';
+                }
+                if (!/[0-9]/.test(val)) {
+                  return 'Password must contain at least one digit';
+                }
```
Comment (kl2karpenko) [inline src/pages/Signup/Signup.tsx:144]: I think it make sence also to check the length of password, so it would be at least I dont know 6-7 symbols

**Review by kl2karpenko — CHANGES_REQUESTED** [review]

File: src/pages/Signup/Signup.tsx (hunk 3 above)
Comment (blystsiv) [reply src/pages/Signup/Signup.tsx:144]: minLength: {
                value: 8,
                message: 'Password must be at least 8 characters long',
              },

**Pull request merged**